includes/Core/Modules/Module.php

```diff
@@ -388,18 +389,36 @@ public function is_connected() {
 	 * @return array List of datapoints.
 	 */
 	final public function get_datapoints() {
-		return array_keys( $this->get_datapoint_services() );
+		$keys        = array();
+		$definitions = $this->get_datapoint_definitions();
+
+		foreach ( array_keys( $definitions ) as $key ) {
+			$parts = explode( ':', $key );
+			$name  = end( $parts );
+			if ( ! empty( $name ) ) {
+				$keys[ $name ] = $name;
+			}
+		}
+
+		return array_values( $keys );
 	}
 
 	/**
 	 * Returns the mapping between available datapoints and their services.
 	 *
 	 * @since 1.0.0
 	 * @since 1.9.0 No longer abstract.
+	 * @deprecated n.e.x.t
 	 *
 	 * @return array Associative array of $datapoint => $service_identifier pairs.
 	 */
 	protected function get_datapoint_services() {
+		_deprecated_function(
+			sprintf( '%s::get_datapoint_services', static::class ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			'',
+			sprintf( '::get_datapoint_definitions', static::class ) // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
* The first argument can simply be `__METHOD__`
* The second argument here should be `n.e.x.t`
* Missing placeholder in template string in the 3rd argument

This can be simplified a bit into a single line similar to this without `sprintf`
https://github.com/google/site-kit-wp/blob/3590c624a075e9d9cb28bc9bf6ee313dd7aba516/includes/Core/Authentication/Clients/OAuth_Client.php#L787

Reply:
Updated.